File: Treasury.Domain/Model/Frappeal.cs
﻿using System;
using System.Collections.Generic;

#nullable disable

namespace Treasury.Domain.Model
{
    public partial class Frappeal
    {
        public int Id { get; set; }
        public int FrId { get; set; }
        public string NewDotNumber { get; set; }
        public DateTime AppealDate { get; set; }
        public string Description { get; set; }
        public decimal AppealAmount { get; set; }
        public string Decision { get; set; }
        public decimal ApprovedAppeal { get; set; }
        public string Notes { get; set; }
        public string MinutesLink { get; set; }
        public DateTime Timestamp { get; set; }

        public virtual FundingRequest Fr { get; set; }
    }
}

File: src/main/java/edu/wpi/sga/treasury/domain/model/FRAppeal.java
package edu.wpi.sga.treasury.domain.model;

import lombok.Getter;
import lombok.Setter;

import javax.persistence.*;
import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.LocalDateTime;

@Getter
@Setter
@Entity
@Table(name = "fr_appeal")
public class FRAppeal {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id", nullable = false)
    private Integer id;

    @OneToOne(optional = false)
    @JoinColumn(name = "funding_request_id", nullable = false)
    private FundingRequest fundingRequest;

    @Column(name = "new_dot_number", nullable = false, length = 6)
    private String newDotNumber;

    @Column(name = "appeal_date", nullable = false)
    private LocalDate appealDate;

    @Column(name = "description")
    private String description;

    @Column(name = "appeal_amount", nullable = false, precision = 10, scale = 2)
    private BigDecimal appealAmount;

    @Column(name = "decision", nullable = false, length = 20)
    private String decision;

    @Column(name = "approved_appeal", nullable = false, precision = 10, scale = 2)
    private BigDecimal approvedAppeal;

    @Column(name = "notes")
    private String notes;

    @Column(name = "minutes_link")
    private String minutesLink;

    @Column(name = "last_modified", nullable = false)
    private LocalDateTime lastModified;

}